### PR TITLE
feat: select component growth enabled

### LINKED ISSUE #1757: Bug: Text overflowing in time zone dropdown field
### Describe the bug

**Current Behavior**
When a new user logs in they are treated with this page. 
image

As you can see in the Time Zone dropdown field the selected field in the dropdown is overflowing. 
image

**Suggested Solution**
The dropdown field should expand accordingly.

If possible, I would love to be assigned to work on this task.

### Steps to reproduce

1. Connect with Github from a new account
image

2. Click on **Complete the onboarding** 
image

3. Authenticate with Github

4. Choose interests

After following those steps you should be able to reach to this page
image

### Browsers

Edge

### Additional context (Is this in dev or production?)

Dev

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by Hiperultimate**
## Description

The text in Time Zone* using Select component had spacing issues when larger strings were used. (Please check the video added in the recordings section)

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [x] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
Fixes #1757 

## Mobile & Desktop Screenshots/Recordings

https://github.com/open-sauced/app/assets/32923170/95a68d6f-6150-4aa3-968d-0d5fc68c55f1

Before (Mobile)
image

image

image

image

After (Mobile)
image

image

image

image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?
I was only able to check one other place where Select component was used. If possible, please check other places where it was used to we can confirm that other pages are acting as they are meant to be. Or just point me in the right direction so I can check it.

## [optional] What gif best describes this PR or how it makes you feel?
O_O

### CODE REVIEW

File: components/atoms/Select/select.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ const SelectTrigger = React.forwardRef<
   <SelectPrimitive.Trigger
     ref={ref}
     className={clsx(
-      "flex h-[2.1rem] w-full items-center justify-between rounded-lg overflow-hidden   px-3 py-2 text-sm border  focus:outline-none  disabled:cursor-not-allowed disabled:opacity-50",
+      "flex min-h-[2.1rem] w-full items-center justify-between rounded-lg overflow-hidden  px-3 py-[5.8px] text-sm border  focus:outline-none  disabled:cursor-not-allowed disabled:opacity-50",
```
Comment (bdougie) [inline components/atoms/Select/select.tsx:22]: Hardcoding pixels `py-[5.8px]` fixes this page, but I am curious if it breaks others.

Comment (Hiperultimate) [reply components/atoms/Select/select.tsx:22]: Yes, I am also skeptical about that. I made sure the height remains the same if the input value is not making the whole Select component occupy two lines or more; or if its empty. 
I have searched in the application and checked one other place where Select component was being used and compared it with the older implementation and that was working fine as well. Talking about the settings page.
If anyone can point me to places where it is being used I can check it myself and add some pictures so we can confirm it.

**Comment by OgDev-01** [comment]
Thanks for this PR @Hiperultimate, I'm not sure this looks great visually... I suggest we truncate the value when it is longer than the width of the select
Screenshot 2023-09-28 at 19 06 53

We can start by leveraging the approach stated in https://github.com/open-sauced/app/issues/1763 to truncate the value

**Comment by OgDev-01** [comment]
Hey @Hiperultimate, feel free to @mention any of the @open-sauced/enginering member once the changes are made and this is ready for review 👍

**Comment by OgDev-01** [comment]
Hey @Hiperultimate, checking in to see how this is going... any blockers?

**Comment by Hiperultimate** [comment]
Hey @OgDev-01, so sorry for the late reply. I did try fixing the styling through tailwindcss but nothing seemed to work and I got a little busy after that so I couldnt spend more time on this.

**Comment by OgDev-01** [comment]
Hey @Hiperultimate, thanks for letting us know the current state of this issue... I'll take it from here to finish. Thanks one's again for attempting 👍. feel free to take a look at other issues you can take on when you have the time 🍕

**Comment by OgDev-01** [comment]
This is ready for review cc @open-sauced/engineering


https://github.com/open-sauced/app/assets/62995161/712b2ef3-a89e-4cfb-9992-417db5f998c3

### Note on the fix i applied  cc: @Hiperultimate 

While debugging to see why the tailwind classes weren’t working, i discovered radix adds a span element automatically to the trigger that carries the selected value... I had to manually style the span to truncate the text relative to its wrapper (trigger). Link to how i discovered truncating it relation to wrapper width  https://css-tricks.com/snippets/css/truncate-string-with-ellipsis/

**Comment by OgDev-01** [comment]
@open-sauced/engineering , This PR is ready for review

File: components/atoms/Select/select.tsx [hunk 2]
```diff
@@ -19,7 +19,7 @@ const SelectTrigger = React.forwardRef<
   <SelectPrimitive.Trigger
     ref={ref}
     className={clsx(
-      "flex h-[2.1rem] w-full items-center justify-between rounded-lg overflow-hidden   px-3 py-2 text-sm border  focus:outline-none  disabled:cursor-not-allowed disabled:opacity-50",
+      "flex min-h-[2.1rem] w-full items-center radix-select-trigger justify-between rounded-lg overflow-hidden  px-3 py-[5.8px] text-sm border  focus:outline-none  disabled:cursor-not-allowed disabled:opacity-50",
```
Comment (brandonroberts) [inline components/atoms/Select/select.tsx:22]: Why is the 5.8px so specific?

```suggestion
      "flex min-h-[2.1rem] w-full items-center radix-select-trigger justify-between rounded-lg overflow-hidden  px-3 py-[5.8px] text-sm border  focus:outline-none  disabled:cursor-not-allowed disabled:opacity-50",
```

Comment (OgDev-01) [reply components/atoms/Select/select.tsx]: Must be an oversight 👀

**Review by brandonroberts — APPROVED** [review]

File: components/molecules/ComponentDateFilter/component-date-filter.tsx [hunk 3]
```diff
@@ -22,15 +22,15 @@ const ComponentDateFilter = ({ setRangeFilter, defaultRange }: ComponentDateFilt
   return (
     <div className="flex items-center text-sm bg-white rounded-lg shrink-0 w-max">
       {dates.map((range, index) => (
-        <div
+        <button
```
Comment (nickytonline) [inline components/molecules/ComponentDateFilter/component-date-filter.tsx:25]: Noice!

**Review by nickytonline — APPROVED** [review]
:shipit:

File: components/molecules/ComponentDateFilter/component-date-filter.tsx (hunk 3 above)
Comment (OgDev-01) [reply components/molecules/ComponentDateFilter/component-date-filter.tsx:25]: Was intentional 😀.

**Pull request merged**